app/controllers/http_proxies_controller.rb

```diff
@@ -25,6 +25,10 @@ def edit
 
   def test_connection
     http_proxy = HttpProxy.new(http_proxy_params)
```

Comment:
> It may error out if the name is not specified though, which not be what you want.

As I predicted, this is indeed something that does show up and can block users creating a new proxy.

Would it make sense to have some default? Perhaps this works:
```suggestion
    http_proxy = HttpProxy.new({name: 'dummy'}.update(http_proxy_params))
```

I was playing with writing some integration tests for this, but I don't have NPM set up right now. I would like to see some tests though.

Reply:
Sounds good, will go with this default, and start working on some tests.